lib/destination/dml/merge.go

```diff
@@ -307,3 +307,91 @@ func (m *MergeArgument) GetStatement() (string, error) {
 			Prefix:    "cc.",
 		})), nil
 }
+
+func (m *MergeArgument) GetMSSQLStatement() (string, error) {
+	if err := m.Valid(); err != nil {
+		return "", err
+	}
+
+	var idempotentClause string
+	if m.IdempotentKey != "" {
+		idempotentClause = fmt.Sprintf("AND cc.%s >= c.%s ", m.IdempotentKey, m.IdempotentKey)
+	}
+
+	var equalitySQLParts []string
+	for _, primaryKey := range m.PrimaryKeys {
+		// We'll need to escape the primary key as well.
+		equalitySQL := fmt.Sprintf("c.%s = cc.%s", primaryKey.EscapedName(), primaryKey.EscapedName())
+		equalitySQLParts = append(equalitySQLParts, equalitySQL)
+	}
+
+	cols := m.ColumnsToTypes.GetColumnsToUpdate(*m.UppercaseEscNames, &sql.NameArgs{
+		Escape:   true,
+		DestKind: m.DestKind,
+	})
+
+	if m.SoftDelete {
+		return fmt.Sprintf(`
+			MERGE INTO %s c using %s as cc on %s
+				when matched %sthen UPDATE
```

Comment:
Nit: uppercase `using`, `when`, and `then`.

Reply:
Adding TODO for a FF